script/licenses.py

```diff
@@ -244,8 +242,88 @@ def main(argv: Sequence[str] | None = None) -> int:
     return exit_code
 
 
+def extract_license_classifier(classifiers: list[str] | None) -> list[str]:
+    """Extract license from list of classifiers."""
+    return [
+        license_classifier
+        for classifier in classifiers or ()
+        if classifier.startswith("License")
+        and (license_classifier := classifier.rpartition(" :: ")[2])
+        and license_classifier != "OSI Approved"
+    ]
+
+
+def get_package_metadata(dist: metadata.Distribution) -> PackageMetadata:
+    """Get package metadata for distribution."""
+    return {
+        "name": dist.name,
+        "version": dist.version,
+        "license_expression": dist.metadata.get("license-expression"),
+        "license_metadata": dist.metadata.get("license"),
+        "license_classifier": extract_license_classifier(
+            dist.metadata.get_all("classifier")
+        ),
+    }
+
+
+def extract_licenses(args: ExtractArgs) -> int:
```

Comment:
It's easier to read from top to bottom when putting the entry point function at the top and then following with the helper functions. The code doesn't seem to require another order.